impl/src/main/java/io/kroxylicious/testing/kafka/common/KafkaClusterConfig.java

```diff
@@ -360,18 +352,19 @@ public ConfigHolder generateConfigForSpecificNode(KafkaEndpoints kafkaEndpoints,
         return configHolder;
     }
 
-    @NonNull
-    private ConfigHolder configureController(KafkaEndpoints kafkaEndpoints, int nodeId, Properties nodeConfiguration) {
-        return new ConfigHolder(nodeConfiguration, null, null, null,
-                nodeId, kafkaKraftClusterId);
+    private String formatListeners(Map<String, String> listeners, String listenerNameAddressSeparator) {
+        return listeners.entrySet().stream().map(e -> e.getKey() + listenerNameAddressSeparator + e.getValue()).collect(Collectors.joining(","));
     }
 
     @NonNull
-    private ConfigHolder configureBroker(KafkaEndpoints kafkaEndpoints, int nodeId, TreeMap<String, String> protocolMap, TreeMap<String, String> listeners,
-                                         TreeMap<String, String> advertisedListeners, TreeSet<String> earlyStart, Properties nodeConfiguration) {
-        var interBrokerEndpoint = kafkaEndpoints.getEndpointPair(Listener.INTERNAL, nodeId);
-        var clientEndpoint = kafkaEndpoints.getEndpointPair(Listener.EXTERNAL, nodeId);
-        var anonEndpoint = kafkaEndpoints.getEndpointPair(Listener.ANON, nodeId);
+    private ConfigHolder configureBroker(KafkaListenerSource kafkaListenerSource, ConfigHolder configHolder, TreeMap<String, String> protocolMap,
+                                         TreeMap<String, String> listeners,
+                                         TreeMap<String, String> advertisedListeners, TreeSet<String> earlyStart) {
+        var nodeId = configHolder.nodeId();
+        var nodeConfiguration = configHolder.properties();
+        var interBrokerEndpoint = kafkaListenerSource.getKafkaListener(Listener.INTERNAL, nodeId);
+        var clientEndpoint = kafkaListenerSource.getKafkaListener(Listener.EXTERNAL, nodeId);
+        var anonEndpoint = kafkaListenerSource.getKafkaListener(Listener.ANON, nodeId);
```

Comment:
slightly confused by this as we seem to be resolving each listener but not actually using them.

Reply:
no, all three endpoint threes are used later in the method.   The naming is out of date.  I'll fix that.